.github/workflows/test-python.yaml

```diff
@@ -0,0 +1,133 @@
+name: Test Python
+on:
+  workflow_call:
+    inputs:
+      unit-platforms:
+        required: false
+        type: string
+        default: '["ubuntu-22.04", "ubuntu-24.04", "windows-latest", "macos-latest"]'
```

Comment:
Even just for running tests? I know critical stuff (package, etc.) we want to do self-hosted, but do we have sufficient runner capacity for running tests?

Reply:
Adjusted per separate conversation. We want to use self-hosted runners everywhere it's possible. However we don't have Windows or MacOS self-hosted runners. I've excluded Windows and MacOS to make them opt-in since they're also some of the more expensive runners.